MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/MapGestureDetector.java

```diff
@@ -312,7 +314,11 @@ boolean onGenericMotionEvent(MotionEvent event) {
           float scrollDist = event.getAxisValue(MotionEvent.AXIS_VSCROLL);
 
           // Scale the map by the appropriate power of two factor
-          transform.zoomBy(scrollDist, new PointF(event.getX(), event.getY()));
+          float horizontalScroll = event.getX();
+          if (!uiSettings.isHorizontalScrollGesturesEnabled()) {
+            horizontalScroll = 0;
+          }
+          transform.zoomBy(scrollDist, new PointF(horizontalScroll, event.getY()));
```

Comment:
I think this can be reverted. This is only for mouse-wheel zoom and the `event.getX()` here is the focal point.